Sources/HummingbirdXCT/HBXCTRouter.swift

```diff
@@ -17,41 +17,54 @@ import NIOCore
 import NIOPosix
 
 /// Test sending values to requests to router. This does not setup a live server
-struct HBXCTRouter: HBXCT {
+struct HBXCTRouter: HBXCTApplication {
     /// Dummy request context
     struct RequestContext: HBRequestContext {
         let eventLoop: EventLoop
         var allocator: ByteBufferAllocator { ByteBufferAllocator() }
         var remoteAddress: SocketAddress? { return nil }
     }
 
-    init() {
-        #if os(iOS)
-        self.eventLoopGroup = NIOTSEventLoopGroup()
-        #else
-        self.eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: System.coreCount)
-        #endif
+    let eventLoopGroup: EventLoopGroup
+    let context: HBApplication.Context
+    let responder: HBResponder
+
+    init(builder: HBApplicationBuilder) {
+        self.eventLoopGroup = builder.eventLoopGroup
+        self.context = HBApplication.Context(
+            threadPool: builder.threadPool,
+            configuration: builder.configuration,
+            logger: builder.logger,
+            encoder: builder.encoder,
+            decoder: builder.decoder
+        )
+        self.responder = builder.router.buildRouter()
+    }
+
+    func shutdown() throws {
```

Comment:
availability `noasync` right?

Reply:
Made it async, given it was being called from an async context